src/storage_utils.cpp

```diff
@@ -607,7 +607,7 @@ namespace libtorrent { namespace aux {
 		int ret = 0;
 		for (auto buf : bufs)
 		{
-			ret += buf.size();
+			ret += static_cast<int>(buf.size());
```

Comment:
I think `aux::numeric_cast<>` might be more appropriate here, as that also contains an assert that there is no overflow

Reply:
perhaps, I haven't really given thought to the overflow...

BTW the TORRENT_ASSERT looks weird there, shouldn't it be using `&&` instead of `||`?
```c++
TORRENT_ASSERT(std::is_unsigned<In>::value && std::is_signed<T>::value && std::int64_t(v) >= 0);
TORRENT_ASSERT(std::is_signed<In>::value && std::is_unsigned<T>::value && std::size_t(v) <= std::size_t((std::numeric_limits<T>::max)()));
```
https://github.com/arvidn/libtorrent/blob/RC_1_2/include/libtorrent/aux_/numeric_cast.hpp#L49-L52

Reply:
Never mind...